src/components/switchTheme/switchTheme.tsx

```diff
@@ -1,21 +1,22 @@
 import { IconicButton } from 'src/components/_common/iconicButton/iconicButton';
 import { SwitchThemeIcon } from 'src/components/_common/icons/switchThemeIcon';
+import { useTheme } from 'src/components/switchTheme/hook';
 
-import { useSwitchTheme } from './hook';
 import { useSelector } from 'react-redux';
 import { selectMenu } from 'src/store/selectors/menuButton';
 
 export const SwitchTheme = () => {
-  const handleSwitchTheme = useSwitchTheme();
+  const { theme, toggleTheme } = useTheme();
+
+  const handleToggle = () => {
+    toggleTheme();
+  };
+
   const isMenuActive = useSelector(selectMenu);
 
   return (
     <>
-      {isMenuActive && (
-        <IconicButton onClickHandler={handleSwitchTheme}>
-          <SwitchThemeIcon />
-        </IconicButton>
-      )}
+      {isMenuActive && <IconicButton onClickHandler={handleToggle}>{theme ? <SwitchThemeIcon /> : false}</IconicButton>}
```

Comment:
only I think now, why we have this button not here)

Reply:
The button is conditionally rendered based on the isMenuActive state because it is only meant to be visible when the menu is active. This behavior is intentional, as it ensures that the button appears only when the menu is in an "active" state, aligning with the desired functionality.

If the button were to always be rendered, it could lead to unnecessary UI elements being displayed when the menu is not active, which might not be the intended design. So, the current implementation is correct as per the functionality requirements.

Reply:
ok, forget it)